src/module_PI/Raspberry_PI/main/EmailNotifier.java

```diff
@@ -0,0 +1,239 @@
+package module_PI.Raspberry_PI.main;
+
+import java.io.*;
+import java.util.*;
+
+import javax.activation.*;
+import javax.mail.*;
+import javax.mail.internet.*;
+
+public class EmailNotifier {
+
+	private String gmailUsername;
+	private String gmailPassword;
+
+	private String recipientEmailAddress;
+	private String emailSubject = "Calico Home Security";
+	private String emailMessage;
+	private String attachmentFilePath;
+
+	private Properties mailSessionProperties;
+	private Session javaMailSession;
+	private Message javaMailMessage;
+	private BodyPart messageBodyPart;
+	private Multipart multipart;
+	private DataSource attachmentDataSource;
+
+	public EmailNotifier(String recipientEmailAddress, String message) {
+		this(recipientEmailAddress, message, null);
+	}
+
+	public EmailNotifier(String recipientEmailAddress, String message, String pathToAttachment) {
+		this.loadGmailCredentials();
+		this.setRecipientEmailAddress(recipientEmailAddress);
+		this.emailMessage = message;
+		this.setAttachmentFilePath(pathToAttachment);
+		buildPropertiesAndCreateSession();
+	}
+
+	private void loadGmailCredentials() {
+		try {
+			Properties gmailProperties = new Properties();
+			FileInputStream in = new FileInputStream("gmailCredentials");
```

Comment:
Reading that article it seems like you still have to save the credentials to a properties file, like I was doing. Maybe I'm missing something or maybe I'm misunderstanding, but my next fix keeps writing to a properties file. I'm having a hard time understanding the visibility of the properties file. Like, if I create a local variable called gmailProperties in one method and try to call it from another class's method, it can't see it, which is how I understand Java, but doesn't mesh with how I understand your description of Java Properties. If I'm misunderstanding, please clarify. I appreciate your feedback.